trino/client.py

```diff
@@ -839,3 +768,147 @@ def decorated(*args, **kwargs):
         return decorated
 
     return wrapper
+
+
+class RowMapperFactory:
+    """
+    Given the 'columns' result from Trino, generate a list of
+    lambda functions (one for each column) which will process a data value
+    and returns a RowMapper instance which will process rows of data
+    """
+
+    def create(self, columns, experimental_python_types):
+        assert columns is not None
+
+        if experimental_python_types:
+            return RowMapper([self._col_func(column['typeSignature']) for column in columns])
+        return RowMapper()
+
+    def _col_func(self, column):
+        col_type = column['rawType']
+
+        if col_type == 'array':
+            return self._array_map_func(column)
+        elif col_type == 'row':
+            return self._row_map_func(column)
+        elif col_type == 'map':
+            return self._map_map_func(column)
+        elif col_type.startswith('decimal'):
+            return lambda val: Decimal(val)
+        elif col_type.startswith('double') or col_type.startswith('real'):
+            return self._double_map_func()
+        elif col_type.startswith('timestamp'):
+            return self._timestamp_map_func(column, col_type)
+        elif col_type.startswith('time'):
+            return self._time_map_func(column, col_type)
+        elif col_type == 'date':
+            return lambda val: datetime.strptime(val, '%Y-%m-%d').date()
```

Comment:
Seems this won't be able to handle negative years.

Reply:
Python does not seem to handle such years.